astropy/io/fits/fitstime.py

```diff
@@ -0,0 +1,380 @@
+
+# The following is based on the FITS WCS Paper IV, "Representations of time
+# coordinates in FITS".
+# http://adsabs.harvard.edu/abs/2015A%26A...574A..36R
+
+
+# FITS WCS standard specified "4-3" form for non-linear coordinate types
+TCTYP_RE_TYPE = re.compile(r'(?P<type>[A-Z]+)[-]+')
+TCTYP_RE_ALGO = re.compile(r'(?P<algo>[A-Z]+)\s*')
+
+# FITS Time standard specified time units
+FITS_TIME_UNIT = ['s', 'd', 'a', 'cy', 'min', 'h', 'yr', 'ta', 'Ba']
+
+# Global time reference coordinate keywords
+TIME_KEYWORDS = {'TIMESYS': 'scale', 'MJDREF': 'ref_mjd',
+                 'JDREF': 'ref_jd', 'DATEREF': 'ref_date',
+                 'TREFPOS': 'pos', 'TREFDIR': 'dir',
+                 'TIMEUNIT': 'unit', 'TIMEOFFS': 'offs',
+                 'OBSGEO-X': 'loc_x', 'OBSGEO-Y': 'loc_y',
+                 'OBSGEO-Z': 'loc_z', 'DATE': 'date',
+                 'DATE-OBS': 'date-obs', 'DATE-AVG': 'date-avg',
+                 'DATE-BEG': 'date-beg', 'DATE-END': 'date-end',
+                 'MJD-OBS': 'mjd-obs', 'MJD-AVG': 'mjd-avg',
+                 'MJD-BEG': 'mjd-beg', 'MJD-END': 'mjd-end'}
+
+
+# Column-specific time override keywords
+COLUMN_TIME_KEYWORDS = {'TCTYP': 'scale',
+                        'TCUNI': 'unit',
+                        'TRPOS': 'pos'}
+
+
+COLUMN_TIME_KEYWORD_REGEXP = '({0})[0-9]+'.format(
+    '|'.join(COLUMN_TIME_KEYWORDS.keys()))
+
+
+def is_time_column_keyword(keyword):
+    """
+    Check if the FITS header keyword is a time column-specific keyword.
+
+    Parameters
+    ----------
+    keyword : str
+        FITS keyword.
+    """
+    return re.match(COLUMN_TIME_KEYWORD_REGEXP, keyword) is not None
+
+
+# Set astropy time global information
+GLOBAL_TIME_INFO = {'TIMESYS': ('UTC', 'Default time scale'),
+                    'JDREF': (0.0, 'Time columns are jd = jd1 + jd2'),
+                    'TREFPOS': ('TOPOCENTER', 'Time reference position')}
+
+
+def _verify_global_info(global_info):
+    """
+    Given the global time reference frame information, verify that
+    each global time coordinate attribute will be given a valid value.
+
+    Parameters
+    ----------
+    global_info : dict
+        Global time reference frame information.
+    """
+
+    # Translate FITS deprecated scale into astropy scale, or else just convert
+    # to lower case for further checks.
+    global_info['scale'] = FITS_DEPRECATED_SCALES.get(global_info['scale'],
+                                                      global_info['scale'].lower())
+
+    # Verify global time scale
+    if not global_info['scale'] in Time.SCALES:
+        if global_info['scale'] in ['gps', 'local']:
+            warnings.warn(
+                'Global time scale (TIMESYS) has a FITS recognized time scale '
+                'value {!r}. However, since it is not a valid astropy time scale, '
+                'it will not be considered.'.format(global_info['scale']),
+                AstropyUserWarning)
+            global_info['scale'] = None
```

Comment:
The standard states that ``(Global Positioning System): runs (approximately) synchronously with TAI; GPS ≈ TAI − 19 s`` So should we use tai as a scale here?
For ``LOCAL``, we could use UTC since it is the Default time scale for FITS.